core/shared/src/main/scala/scalaz/zio/UIO.scala

```diff
@@ -0,0 +1,534 @@
+   * because a defect has been detected in the code.
+   */
+  final def dieMessage(message: String): UIO[Nothing] = ZIO.dieMessage(message)
+
+  /** Imports a total synchronous effect into a pure `ZIO` value.
+   * The effect must not throw any exceptions. If you wonder if the effect
+   * throws exceptions, then do not use this method, use [[Task.effect]],
+   * [[IO.effect]], or [[ZIO.effect]].
+   *
+   * {{{
+   * val nanoTime: UIO[Long] = IO.effectTotal(System.nanoTime())
+   * }}}
+   */
+  final def effectTotal[A](effect: => A): UIO[A] = ZIO.effectTotal(effect)
+
+  /** Imports a total synchronous effect into a pure `ZIO` value. This variant
+   * of `effectTotal` lets the impure code use the platform capabilities.
+   *
+   * The effect must not throw any exceptions. If you wonder if the effect
+   * throws exceptions, then do not use this method, use [[Task.effect]],
+   * [[IO.effect]], or [[ZIO.effect]].
+   *
+   * {{{
+   * val nanoTime: UIO[Long] = IO.effectTotal(System.nanoTime())
+   * }}}
+   */
+  final def effectTotalWith[A](effect: Platform => A): UIO[A] = ZIO.effectTotalWith(effect)
+
+  /** Returns an effect that yields to the runtime system, starting on a fresh
+   * stack. Manual use of this method can improve fairness, at the cost of
+   * overhead.
+   */
+  final val yieldNow: UIO[Unit] = ZIO.Yield
+
+  /** Returns an effect that forks all of the specified values, and returns a
+   * composite fiber that produces a list of their results, in order.
+   */
+  final def forkAll[A](as: Iterable[ZIO[UioR, UioE, A]]): ZIO[UioR, Nothing, Fiber[UioE, List[A]]] =
+    ZIO.forkAll(as)
+
+  /** Returns an effect that forks all of the specified values, and returns a
+   * composite fiber that produces unit. This version is faster than [[forkAll]]
+   * in cases where the results of the forked fibers are not needed.
+   */
+  final def forkAll_[A](as: Iterable[ZIO[UioR, UioE, A]]): ZIO[UioR, Nothing, Unit] =
+    ZIO.forkAll_(as)
+
+  /** Returns an effect from a [[scalaz.zio.Exit]] value.
+   */
+  final def done[A](r: Exit[UioE, A]): IO[UioE, A] = ZIO.done(r)
+
+  /** Enables supervision for this effect. This will cause fibers forked by
+   * this effect to be tracked and will enable their inspection via [[ZIO.children]].
+   */
+  final def supervised[A](zio: ZIO[UioR, UioE, A]): ZIO[UioR, UioE, A] =
+    ZIO.supervised(zio)
+
+  /** Returns an effect that supervises the specified effect, ensuring that all
+   * fibers that it forks are interrupted as soon as the supervised effect
+   * completes.
+   */
+  final def supervise[A](zio: ZIO[UioR, UioE, A]): ZIO[UioR, UioE, A] =
+    ZIO.supervise(zio)
+
+  /** Returns an effect that supervises the specified effect, ensuring that all
+   * fibers that it forks are passed to the specified supervisor as soon as the
+   * supervised effect completes.
+   */
+  final def superviseWith[A](
+    zio: ZIO[UioR, UioE, A]
```

Comment:
Ditto for all these, they become much simpler, e.g. `UIO[A]` in the above example.